### PR TITLE
Development/sorted classes and style

### LINKED ISSUE #126: Support Sorted Class

### PR DESCRIPTION
**Pull request by Chun-MingChen**
fix #126 and #127 

Hi @AllenFang, the PR aims to support sorted `classes` and `style`. Please take a look, thanks :)

* support sorted `classes` and `style`.
* typo fix

### CODE REVIEW

**Comment by AllenFang** [comment]
@Chun-MingChen I just wondering why you have following design:
```js
sortedHeader: {
  classes: ....
  styles: ....
}
```

Our cell, row, selection and cell edit usually separate classes and styles as two different prop

**Comment by Chun-MingChen** [comment]
@AllenFang 

Sure it does be inconsistent. I've fixed it to separate `sortingClass` and `sorting style` into two props. Thanks 👍

File: docs/README.md [hunk 1]
```diff
@@ -101,4 +103,31 @@ const defaultSorted = [{
...
+```js
+const sortingHeaderClasses = function callback(column, colIndex) { ... }
+```
+
+* column: The value of current column.
+* colIndex: The index of the current column being processed in BootstrapTable.
+
+### <a name='sortingHeaderStyle'>sortingHeaderStyle - [Object | Function]</a>
+
+It's similiar to `sortingHeaderClasses`. It allows to customize style of `header cell` which is sorting based on. `Object` and `callback` function are acceptable. `callback` takes `2` arguments and an `Object` is expected to return:
```
Comment (AllenFang) [inline docs/README.md]: An style object or callback function are acceptable

File: docs/README.md [hunk 2]
```diff
@@ -101,4 +103,31 @@ const defaultSorted = [{
...
+`sortingHeaderClasses` allows you to customize `class` for `header cell` which the table was sorting based on. 
+
+```js
+const sortingHeaderClasses = 'demo-sorting';
+```
+
+Furthermore, it also accepts a callback function which takes `2` arguments and `String` is expected to return:
+
+```js
+const sortingHeaderClasses = function callback(column, colIndex) { ... }
```
Comment (AllenFang) [inline docs/README.md]: es6 style

File: docs/README.md [hunk 3]
```diff
@@ -101,4 +103,31 @@ const defaultSorted = [{
...
+```
+
+* column: The value of current column.
+* colIndex: The index of the current column being processed in BootstrapTable.
```
Comment (AllenFang) [inline docs/README.md]: Additional pass sort order to callback function as third argument

**Comment by Chun-MingChen** [comment]
Hi @AllenFang: 

Sorry for commit late. After our discussion, I implemented header sorting classes and style in column level. Furthermore, `isLastSorting` indicates that the field was last sorting or not. Please take a look, thanks :)

File: docs/columns.md [hunk 4]
```diff
@@ -419,6 +421,35 @@ A new `Object` will be the result of element headerAttrs.
 > Same as [column.attrs](#attrs), it has lower priority and will be
 > overwrited when other props related to HTML attributes were given.
 
+### <a name='headerSortingClasses'>headerSortingClasses - [String | Function]</a>
+
+`headerSortingClasses` allows to customize `class` for header cell which the table was sorting based on.
```
Comment (AllenFang) [inline docs/columns.md]: `headerSortingClasses` allows to customize `class` for header cell when this column is sorting

File: docs/columns.md [hunk 5]
```diff
@@ -419,6 +421,35 @@ A new `Object` will be the result of element headerAttrs.
...
+
+Furthermore, it also accepts a callback which takes **4** arguments and `String` is expected to return:
+
+```js
+const headerSortingClasses = (column, sortOrder, isLastSorting, colIndex) => { ... }
+```
+
+* `column`: The value of current column.
+* `sortOrder`: The order of current sorting
+* `isLastSorting`: Is the last one of sorting.
```
Comment (AllenFang) [inline docs/columns.md]: Is the last one of sorted columns.

File: docs/columns.md [hunk 6]
```diff
@@ -419,6 +421,35 @@ A new `Object` will be the result of element headerAttrs.
...
+* `colIndex`: The index of the current column being processed in BootstrapTable.
+
+### <a name='headerSortingStyle'>headerSortingStyle - [Object | Function]</a>
+
+It's similiar to [headerSortingClasses](#headerSortingClasses). It allows to customize the style of header cell which the table was sorting based on. A style `Object` and `callback` are acceptable. `callback` takes **4** arguments and an `Object` is expected to return:
```
Comment (AllenFang) [inline docs/columns.md]: It allows to customize the style of header cell when this column is sorting

File: packages/react-bootstrap-table2-example/examples/sort/default-sort-table.js [hunk 7]
```diff
@@ -41,7 +41,17 @@ const columns = [{
...
+  dataField: 'name',
+  order: 'desc'
+}];
+
+<BootstrapTable
+  keyField="id"
+  data={ products }
+  columns={ columns }
+  defaultSorted={ defaultSorted } 
+/>
```
Comment (AllenFang) [inline packages/react-bootstrap-table2-example/examples/sort/default-sort-table.js]: Thanks

File: packages/react-bootstrap-table2-example/stories/index.js [hunk 8]
```diff
@@ -122,7 +124,9 @@ storiesOf('Work on Rows', module)
 storiesOf('Sort Table', module)
   .add('Enable Sort', () => <EnableSortTable />)
   .add('Default Sort Table', () => <DefaultSortTable />)
-  .add('Custom Sort Fuction', () => <CustomSortTable />);
+  .add('Custom Sort Fuction', () => <CustomSortTable />)
+  .add('Header sorting classes', () => <HeaderSortingClassesTable />)
```
Comment (AllenFang) [inline packages/react-bootstrap-table2-example/stories/index.js]: **Custom Classes on Sorting Header Column**

File: packages/react-bootstrap-table2-example/stories/index.js [hunk 9]
```diff
@@ -122,7 +124,9 @@ storiesOf('Work on Rows', module)
...
+  .add('Header sorting style', () => <HeaderSortingStyleTable />);
```
Comment (AllenFang) [inline packages/react-bootstrap-table2-example/stories/index.js]: **Custom Style on Sorting Header Column**

File: packages/react-bootstrap-table2/test/header-cell.test.js [hunk 10]
```diff
@@ -436,6 +436,233 @@ describe('HeaderCell', () => {
           });
         });
       });
+
+      describe('when headerSortingClasses was defined ', () => {
```
Comment (AllenFang) [inline packages/react-bootstrap-table2/test/header-cell.test.js]: when headerSortingClasses is defined

File: packages/react-bootstrap-table2/test/header-cell.test.js [hunk 11]
```diff
@@ -436,6 +436,233 @@ describe('HeaderCell', () => {
...
+            wrapper = shallow(
+              <HeaderCell
+                column={ column }
+                index={ index }
+                sorting
+                sortOrder={ order }
+              />);
+          });
+
+          it('should append classes correcly', () => {
```
Comment (AllenFang) [inline packages/react-bootstrap-table2/test/header-cell.test.js]: should + V-ing

File: packages/react-bootstrap-table2/test/header-cell.test.js [hunk 12]
```diff
@@ -436,6 +436,233 @@ describe('HeaderCell', () => {
...
+                  isLastSorting
+                />);
+
+              expect(classesCallBack.callCount).toBe(1);
+              expect(classesCallBack.calledWith(column, order, isLastSorting, index)).toBe(true);
+            });
+          });
+        });
+
+        describe('if column.headerClasses was defined as well', () => {
```
Comment (AllenFang) [inline packages/react-bootstrap-table2/test/header-cell.test.js]: if column.headerClasses is defined as well

File: packages/react-bootstrap-table2/test/header-cell.test.js [hunk 13]
```diff
@@ -436,6 +436,233 @@ describe('HeaderCell', () => {
...
+
+            expect(wrapper.length).toBe(1);
+            expect(wrapper.hasClass('sortable')).toBe(true);
+            expect(wrapper.hasClass(classes)).toBe(true);
+            expect(wrapper.hasClass(column.headerClasses)).toBe(true);
+          });
+        });
+      });
+
+      describe('when headerSortingStyle was defined', () => {
```
Comment (AllenFang) [inline packages/react-bootstrap-table2/test/header-cell.test.js]: is

File: packages/react-bootstrap-table2/test/header-cell.test.js (hunk 11 above)
Comment (Chun-MingChen) [reply packages/react-bootstrap-table2/test/header-cell.test.js]: I thought that will be great for should + VR :)

**Pull request merged**